### DISCUSSION TITLE
How can I get processor(ARMv7/ARMv8/Intel x86) info in ANIMATE ?

### DISCUSSION THREAD

**Discussion by Gokulv617** [airsdk/Adobe-Runtime-Support#717] [Q&A]
```
var appXML:XML =  NativeApplication.nativeApplication.applicationDescriptor;
var ns:Namespace = appXML.namespace();
trace(appXML.ns::versionNumber);   // for getting version number from application descriptor
```

### Same like above, how can I get the processor info in ADOBE ANIMATE ?

![img_20210305](https://user-images.githubusercontent.com/43432782/110067948-b29cfb00-7d9a-11eb-9728-f77c26ca25f9.png)

**Comment by marchbold** [comment]
That information is not in the application descriptor. It is a packaging option. 

You could try Capabilities - cpuAddressSize and cpuArchitecture but I haven't tested that option myself.

https://help.adobe.com/en_US/FlashPlatform/reference/actionscript/3/flash/system/Capabilities.html#version

**Reply by Gokulv617** [reply]
okay. I will try and let you know the result

**Comment by itlancer** [comment]
If you look directory name inside lib directory inside application (you can see it if "unzip" final apk) you can "see" target architecture of apk build. This directory has name according architecture (`armeabi-v7a`, `arm64-v8a`, `x86`, `x86_64`).
I don't know how to get this directory name via AS3 code but found workaround.
I have native extension (you can create it even without any native code) so final apk build will have similar directory inside ANE directory. Inside apk it looks like:
assets/META-INF/AIR/extensions/com.native.extension/META-INF/ANE/**Android-ARM**
assets/META-INF/AIR/extensions/com.native.extension/META-INF/ANE/**Android-ARM64**
assets/META-INF/AIR/extensions/com.native.extension/META-INF/ANE/**Android-x86**
assets/META-INF/AIR/extensions/com.native.extension/META-INF/ANE/**Android-x64**

So you can check current application build architecture via AS3 side via check if target directory name exists:
`ExtensionContext.getExtensionDirectory("com.native.extension").resolvePath("META-INF/ANE/Android-x86").exists`
etc
where "`com.native.extension`" - your native extension ID.
Moreover, you can use ANY (not your own) native extension for that if you use some.

May be `Capabilities.cpuArchitecture` and `Capabilities.cpuAddressSize`, `Capabilities.supports64BitProcesses` return something similar - didn't test it.
